templates/repo/diff/box.tmpl

```diff
@@ -71,7 +71,7 @@
 				<div id="diff-file-boxes" class="sixteen wide column">
 					{{range $i, $file := .Diff.Files}}
 						{{/*notice: the index of Diff.Files should not be used for element ID, because the index will be restarted from 0 when doing load-more for PRs with a lot of files*/}}
-						{{$blobBase := call $.GetBlobByPathForCommit $.BaseCommit $file.OldName}}
+						{{$blobBase := call $.GetBlobByPathForCommit $.BeforeCommit $file.OldName}}
 						{{$blobHead := call $.GetBlobByPathForCommit $.HeadCommit $file.Name}}
```

Comment:
In a similar fashion, shouldn't this be `AfterCommit` then?

Reply:
Most of the variables using this start with "head".

Reply:
Yes, and most of the variables intended for the `base` branch start with `base`?

Reply:
I’ll have to take a look at where AfterCommit comes from. I don’t see how that would be the case though: we should be comparing the “current commit” against the last common ancestor. I expect AfterCommit will just be the same commit ref as BeforeCommit in a three dot diff.

Reply:
I checked it out, and `AfterCommitID` is just set to headCommitID [here](https://github.com/go-gitea/gitea/blob/main/routers/web/repo/compare.go#L576).

Whereas, beforeCommit depends on whether it's a direct comparison (2-dot) or indirect comparison (3-dot) [here](https://github.com/go-gitea/gitea/blob/6221a6fd5450692ae27e5602b41fc9ebd9150736/routers/web/repo/compare.go#L596).

I could change `HeadCommit` to `AfterCommit` in the context, but it's just a matter of naming preference. Maybe the table below will help clarify (in parenthesis is what we name them in the code). In other places in the code we call it "start" and "end".

| use-case                       |  before (from/red) |  after (to/green) |
|-----------------------|-------------------|------------------|
| commit                          |  parent ("[parentCommit](https://github.com/go-gitea/gitea/blob/main/routers/web/repo/commit.go#L325)" ) | commit ("[commit](https://github.com/go-gitea/gitea/blob/main/routers/web/repo/commit.go#L277)") |
| direct compare             | base                       | head |
| indirect compare          | merge-base ("[MergeBase](https://github.com/go-gitea/gitea/blob/main/routers/web/repo/compare.go#L554)")| head |
| PR compare (indirect) | ^                  ("[startCommit](https://github.com/go-gitea/gitea/blob/main/routers/web/repo/pull.go#L708)")  | ^  ("endCommit")|

So, there's probably opportunity for cleanup here, but the behavior should be correct now.